partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java

```diff
@@ -0,0 +1,481 @@
+import org.partiql.ast.v1.graph.GraphQuantifier;
+import org.partiql.ast.v1.graph.GraphRestrictor;
+import org.partiql.ast.v1.graph.GraphSelector;
+import org.partiql.value.PartiQLValue;
+
+import java.util.List;
+import java.util.Map;
+
+// TODO docs for all factory methods
+public class Ast {
+    // Expr
+    @NotNull
+    public static ExprAnd exprAnd(Expr lhs, Expr rhs) {
+        return new ExprAnd(lhs, rhs);
+    }
+
+    @NotNull
+    public static ExprArray exprArray(@NotNull List<Expr> values) {
+        return new ExprArray(values);
+    }
+
+    @NotNull
+    public static ExprBag exprBag(@NotNull List<Expr> values) {
+        return new ExprBag(values);
+    }
+
+    @NotNull
+    public static ExprBetween exprBetween(@NotNull Expr value, @NotNull Expr from, @NotNull Expr to, boolean not) {
+        return new ExprBetween(value, from, to, not);
+    }
+
+    @NotNull
+    public static ExprCall exprCall(@NotNull IdentifierChain function, @NotNull List<Expr> args, @Nullable SetQuantifier setq) {
+        return new ExprCall(function, args, setq);
+    }
+
+    @NotNull
+    public static ExprCase exprCase(@Nullable Expr expr, @NotNull List<ExprCase.Branch> branches, @Nullable Expr defaultExpr) {
+        return new ExprCase(expr, branches, defaultExpr);
+    }
+
+    @NotNull
+    public static ExprCase.Branch exprCaseBranch(@NotNull Expr condition, @NotNull Expr expr) {
+        return new ExprCase.Branch(condition, expr);
+    }
+
+    @NotNull
+    public static ExprCast exprCast(@NotNull Expr value, @NotNull DataType asType) {
+        return new ExprCast(value, asType);
+    }
+
+    @NotNull
+    public static ExprCoalesce exprCoalesce(@NotNull List<Expr> args) {
+        return new ExprCoalesce(args);
+    }
+
+    @NotNull
+    public static ExprExtract exprExtract(@NotNull DatetimeField field, @NotNull Expr source) {
+        return new ExprExtract(field, source);
+    }
+
+    @NotNull
+    public static ExprInCollection exprInCollection(@NotNull Expr lhs, @NotNull Expr rhs, boolean not) {
+        return new ExprInCollection(lhs, rhs, not);
+    }
+
+    @NotNull
+    public static ExprIsType exprIsType(@NotNull Expr value, @NotNull DataType type, boolean not) {
+        return new ExprIsType(value, type, not);
+    }
+
+    @NotNull
+    public static ExprLike exprLike(@NotNull Expr value, @NotNull Expr Pattern, @Nullable Expr escape, boolean not) {
+        return new ExprLike(value, Pattern, escape, not);
+    }
+
+    // This representation will be changed in https://github.com/partiql/partiql-lang-kotlin/issues/1589
+    @NotNull
+    public static ExprLit exprLit(@NotNull PartiQLValue value) {
+        return new ExprLit(value);
+    }
+
+    @NotNull
+    public static ExprMatch exprMatch(@NotNull Expr expr, @NotNull GraphMatch pattern) {
+        return new ExprMatch(expr, pattern);
+    }
+
+    @NotNull
+    public static ExprNot exprNot(@NotNull Expr value) {
+        return new ExprNot(value);
+    }
```

Comment:
(self-review) for PartiQL expression `NOT false`,

example usage from Kotlin
```Kotlin
exprNot(exprLit(boolValue(false)))
```

example usage from Java
```Java
exprNot(exprLit(boolValue(false)));
```